src/pydata_sphinx_theme/toctree.py

```diff
@@ -36,7 +37,25 @@ def add_toctree_functions(
     """Add functions so Jinja templates can add toctree objects."""
 
     @lru_cache(maxsize=None)
-    def generate_header_nav_html(n_links_before_dropdown: int = 5) -> str:
+    def get_id_generator(base_id: str) -> Iterator[str]:
+        for n in count(start=1):
+            if n == 1:
+                yield base_id
+            else:
+                yield f"{base_id}-{n}"
+
+    def create_next_id(base_id: str):
+        """Create an id that is build-time unique.
+
+        The function works by sequentially returning "<base_id>", "<base_id>-2",
+        "<base_id>-3", etc. each time it is called.
+        """
+        return next(get_id_generator(base_id))
+
+    @lru_cache(maxsize=None)
+    def generate_header_nav_html(
+        dropdown_id: str, n_links_before_dropdown: int = 5
+    ) -> str:
```

Comment:
the new parameter `dropdown_id` should come *second* in the list and should have a default value. Otherwise we break backward compatibility for users who may have used this function in their own `init` or `conf` files.

By giving it a default value (`None` makes sense) then you can put the if/else logic inside this function instead of in Jinja, which is also nice because flow-control logic is easier to read in python than in jinja (IMO).